projects/storefrontlib/src/cms-components/checkout/guards/checkout.guard.ts

```diff
@@ -1,27 +1,53 @@
 import { Injectable } from '@angular/core';
 import { CanActivate, Router, UrlTree } from '@angular/router';
 
-import { CheckoutConfig } from '../config/checkout-config';
 import { Observable, of } from 'rxjs';
 import { RoutingConfigService } from '@spartacus/core';
+import { ExpressCheckoutService } from '../services/express-checkout.service';
+import { switchMap } from 'rxjs/operators';
+import { CheckoutConfigService } from '../services/checkout-config.service';
+import { CheckoutStepType } from '../model/checkout-step.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CheckoutGuard implements CanActivate {
+  firstStep$: Observable<UrlTree>;
+
   constructor(
```

Comment:
That is a breaking change. You should keep the `config` and add new services as optional. In places where you use new services make checks if they are provided.

Reply:
sure, it is done in https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/4310/files